addons/phone_validation/models/mail_thread_phone.py

```diff
@@ -94,8 +94,23 @@ def _search_phone_mobile_search(self, operator, value):
         if self._phone_search_min_length and len(value) < self._phone_search_min_length:
             raise UserError(_('Please enter at least 3 characters when searching a Phone/Mobile number.'))
 
-        sql_operator = {'=like': 'LIKE', '=ilike': 'ILIKE'}.get(operator, operator)
+        # try to format number, based on current user's country as a wild guess;
+        # if we can format it, we fallback on a direct search to speedup research
+        # otherwise we use regex-based search on phone fields, hoping to find
+        # something that matches
+        formatted = self.env.user._phone_format(number=value)
+        if formatted:
+            if operator in expression.NEGATIVE_TERM_OPERATORS:
+                return expression.AND([
+                    [('phone_sanitized', '!=', formatted)],
+                    expression.AND([[(phone_field, '!=', formatted)] for phone_field in phone_fields])
+                ])
+            return expression.OR([
+                [('phone_sanitized', '=', formatted)],
+                expression.OR([[(phone_field, '=', formatted)] for phone_field in phone_fields])
+            ])
```

Comment:
:thinking: maybe we should still try the regex if there's no match? For example I could be from "FR" and the number I'm looking for be "BE", if I use "0499000000" it can be successfully formatted in both cases but with different results

Reply:
Issue would be that with 1 match you have that result, with no match you would have other results ... but we could check that phone / mobile holds either formatted, either input for example.